android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/ButtonFactory.java

```diff
@@ -49,23 +59,33 @@
 
         List<View> views = new ArrayList<>(1);
         JSONArray canvasIds = new JSONArray();
-        Button button = new Button(context);
-        LinearLayout.LayoutParams layoutParams = new LinearLayout.LayoutParams(
-                ViewGroup.LayoutParams.MATCH_PARENT,
-                ViewGroup.LayoutParams.WRAP_CONTENT);
-        layoutParams.bottomMargin = context.getResources()
-                .getDimensionPixelSize(R.dimen.extra_bottom_margin);
-        button.setLayoutParams(layoutParams);
-        button.setTextSize(TypedValue.COMPLEX_UNIT_PX, context.getResources()
-                .getDimension(R.dimen.button_text_size));
-        button.setTextColor(context.getResources().getColor(android.R.color.white));
-        button.setHeight(context.getResources().getDimensionPixelSize(R.dimen.button_height));
 
+        Button button = createButton(context);
+
+        // set text properties
         String hint = jsonObject.optString(JsonFormConstants.HINT);
         if (!TextUtils.isEmpty(hint)) {
             button.setText(hint);
+            String textColorStr = jsonObject.optString(BTN_TEXT_COLOR);
+            int textColor = Color.parseColor(textColorStr);
+            int textSize = jsonObject.optInt(BTN_TEXT_SIZE);
+            button.setTextColor(textColor);
+            button.setTextSize(textSize);
```

Comment:
Maybe you should first  check if one provided `BTN_TEXT_SIZE`, if its not provided `textSize` will be `0` so you should check if its `0` first